app_dart/lib/src/request_handling/regular_request_handler.dart

```diff
@@ -0,0 +1,119 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
```

Comment:
This class copies functionalities from `api_request_handler` to handle JSON payload, which could be refactored into a `json_request_handler`. Then refactor to a cleaner hierarchy: `api_request_handler` extends `json_request_handler` extends `request_handler`. And process "no auth" requests via `json_request_handler`.

Reply:
Have changed 'regular' to 'noauth' to make it clear. Yeah, this is one option as well. Since api request involves lots of files and existing APIs, I would prefer to avoid touching it for now. Together with the 'noauth' name change, it is should be clear now.